engine/inc/com/centreon/engine/downtimes/downtime_manager.hh

```diff
@@ -40,7 +40,7 @@ class downtime_manager {
                                           uint64_t downtime_id);
   int check_pending_flex_host_downtime(host* hst);
   int check_pending_flex_service_downtime(service* svc);
-  void add_downtime(downtime* dt) noexcept;
+  void add_downtime(std::shared_ptr<downtime> dt) noexcept;
```

Comment:
you can pass a const std::shared_ptr<downtime> & instead to avoid copy constructor